### PR TITLE
Use less magic in exception macros

### PR DESCRIPTION
**Pull request by mvd-ows**
This changeset removes all hidden logic and assumptions from the exception macros:

- Removes conditional logic in macros.
- Removes hidden assumption about the `actual` argument to `THROW_IF` (that it's a valid Windows error code).
- Removes abbreviations (`GLE` meaning `GetLastError()`).
- Removes all but two macros to ensure logic is explicit at the call site.

---
This change is [Reviewable](https://reviewable.io/reviews/mullvad/windows-libraries/29)

### CODE REVIEW

**Review by dlon — APPROVED** [review]
:lgtm: 

> Reviewed 20 of 20 files at r1.
> *[Reviewable](https://reviewable.io/reviews/mullvad/windows-libraries/29#-:-LzI2RBM2dC2kTSBha56:bg7igkv)* status: :shipit: complete! all files reviewed, all discussions resolved

File: src/libcommon/burstguard.cpp [hunk 1]
```diff
@@ -15,14 +15,17 @@ BurstGuard::BurstGuard(Callback callback, uint32_t burstDuration, uint32_t inter
 {
 	m_stateEvent = CreateEventW(nullptr, TRUE, FALSE, nullptr);
 
-	THROW_GLE_IF(NULL, m_stateEvent, "Create BurstGuard thread state event object");
+	if (NULL == m_stateEvent)
```
Comment (dlon) [inline src/libcommon/burstguard.cpp]: Minor thing, but this could be `nullptr`.

**Review by mvd-ows — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/windows-libraries/29)* status: 19 of 20 files reviewed, 1 unresolved discussion (waiting on @dlon)

___
*[src/libcommon/burstguard.cpp, line 18 at r1](https://reviewable.io/reviews/mullvad/windows-libraries/29#-LzI0ehrDhLNzcc8eDq8-r1-18:-LzI4GC117NYRQTrGPgU:bl8lja6) ([raw file](https://github.com/mullvad/windows-libraries/blob/55027e21b8df89e2412b986590d007854c3bfad1/src/libcommon/burstguard.cpp#L18)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Minor thing, but this could be `nullptr`.
</blockquote></details>

Good point. Fixed.

**Review by dlon — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/mullvad/windows-libraries/29)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**